inc/class-statify-frontend.php

```diff
@@ -91,15 +91,24 @@ public static function track_visit( $is_snippet = false ) {
 			$data['referrer'] = esc_url_raw( $referrer, array( 'http', 'https' ) );
 		}
 
-		/* Relative target url */
+		// Relative target URL.
 		$data['target'] = user_trailingslashit( str_replace( home_url( '/', 'relative' ), '/', $target ) );
 
-		// Trim target url.
+		// Trim target URL, i.e. remove query parameters.
 		if ( $wp_rewrite->permalink_structure ) {
-			$data['target'] = wp_parse_url( $data['target'], PHP_URL_PATH );
+			$parsed_target  = wp_parse_url( $data['target'] );
+			$data['target'] = $parsed_target['path'];
+
+			/*
+			 * Re-add AMP parameter to keep that information (only applicable for JS tracking).
+			 * We can assume "amp" to be the only parameter, as ${canonicalPath} already eliminates other queries.
+			 */
+			if ( $is_snippet && isset( $parsed_target['query'] ) && 'amp/' === $parsed_target['query'] ) {
```

Comment:
what if `amp` is not the last query param?

Reply:
Haven't read the code comment :(

Reply:
Strictly speaking it's not a real query parameter at all, just a suffix we add. `${canonicalPath}?amp` will never result in `?amp` not being the very last 4 characters of the argument.

That's not really nice though. We probably should make this clear and split it off before parsing the URL and check against `str_ends_with( $data['target'], '?amp' )` (or equivalent legacy PHP variant)